### DISCUSSION TITLE
FT.TAGVALS Implementation Proposal

### DISCUSSION THREAD

**Discussion by jonathanc-n** [apache/kvrocks#2560] [Q&A]
@PragmaTwice 

For my implementation for FT.TAGVALS, 

all information is here:
https://redis.io/docs/latest/commands/ft.tagvals/ 

I will create a map data structure under field info to keep track of unique tag values. Each unique tag value will have a reference to how many different times it is called and will be decremented or incremented whenever the same tag is added/removed. If it decrements to zero it will be removed. If multiple tags are being added/removed at the same time, the changes will be written in a batch. 

What are your thoughts for this?

**Comment by PragmaTwice** [comment]
So how do you persist it to disk? What is its encoding format? How do you load it at initialization of Kvrocks? 

Is such tracking efficient? I think if users rarely perform this operation, then precise tracking will be unnecessary.

Also, before contributing to Kvrocks Search, please read:
- https://kvrocks.apache.org/community/kvrocks-search-index-encoding
- https://kvrocks.apache.org/blog/kqir-query-engine

**Reply by jonathanc-n** [reply]
I see  thank you. I doubt users would perform the operation very often, do you think iterating through all the keys when the command is called would be better?
